src/encoded/views/download.py

```diff
@@ -0,0 +1,154 @@
+from collections import OrderedDict
+from pyramid.view import view_config
+from pyramid.response import Response
+from ..embedding import embed
+from urllib.parse import (
+    parse_qs,
+    urlencode,
+)
+
+# includes concatenated properties
+_tsv_mapping = OrderedDict([
+    ('Accession', ('files.accession')),
+    ('Experiment', ('accession')),
+    ('Assay', ('assay_term_name')),
+    ('Biosample term id', ('biosample_term_id')),
+    ('Biosample term name', ('biosample_term_name')),
+    ('Biosample type', ('biosample_type')),
+    ('Biosample life stage', ('replicates.library.biosample.life_stage')),
+    ('Biosample sex', ('replicates.library.biosample.sex')),
+    ('Biosample organism', ('replicates.library.biosample.organism.name')),
+    ('Biosample treatments', ('replicates.library.biosample.treatments.treatment_term_name')),
+    ('Biosample subcellular fraction term name', ('replicates.library.biosample.subcellular_fraction_term_name')),
+    ('Biosample phase', ('replicates.library.biosample.phase')),
+    ('Experiment target', 'target.name'),
+    ('Antibody accession', ('replicates.antibody.accession')),
+    ('Nucleic acid term name', ('replicates.library.nucleic_acid_term_name')),
+    ('Depleted in term name', ('replicates.library.depleted_in_term_name')),
+    ('Library extraction method', ('replicates.library.extraction_method')),
+    ('Library fragmentation method', ('replicates.library.fragmentation_method')),
+    ('Libary lysis method', ('replicates.library.lysis_method')),
+    ('Read length', ('replicates.library.read_length')),
+    ('Paired endedness', ('replicates.library.paired_ended')),
+    ('Experiment date released', ('date_released')),
+    ('Project', ('award.project')),
+    ('File format', ('files.file_format')),
+    ('Size', ('files.file_size')),
+    ('Output type', ('files.output_type')),
+    ('Lab', ('files.lab')),
+    ('md5sum', ('files.md5sum')),
+    ('href', ('files.href')),
+    ('Assembly', ('files.assembly'))
+])
+
+
+def dict_generator(indict, pre=None):
+    pre = pre[:] if pre else []
+    if isinstance(indict, dict):
+        for key, value in indict.items():
+            if isinstance(value, dict):
+                for d in dict_generator(value, [key] + pre):
+                    yield d
+            elif isinstance(value, list) or isinstance(value, tuple):
+                for v in value:
+                    for d in dict_generator(v, [key] + pre):
+                        yield d
+            else:
+                yield pre + [key, value]
+    else:
+        yield indict
+
+
+@view_config(route_name='metadata', request_method='GET', renderer='tsv')
+def metadata_tsv(context, request):
+
+    param_list = parse_qs(request.matchdict['search_params'].encode('utf-8'))
```

Comment:
Any reason not to just use request.params here and supply them as the query string?